test/puppetlabs/puppetdb/cli/services_test.clj

```diff
@@ -546,3 +546,23 @@
                         (set (get-temporal-partitions "reports"))))
                  (is (= (->> event-parts (sort-by :table) (drop 1) set)
                         (set (get-temporal-partitions "resource_events")))))))))))))
+
+(deftest initialize-db
+  (testing "use connection user for database connections"
+    (let [con-mig-user "conn-migration-user-value"
+          mig-pass     "migrator-password-value"
+          user         "user-value"
+          pass         "password-value"
+          validation-fn (fn [config metrics-registry]
+                          (is (= con-mig-user (:connection-migrator-username config)))
+                          (is (= con-mig-user (:user config)))
+                          (is (= mig-pass (:migrator-password config)))
+                          (is (= mig-pass (:password config)))
+                          (throw (Exception.
+                                   "everything ok exception")))]
+      (with-redefs [jdbc/make-connection-pool validation-fn]
+        (testing "should use connection user"
+          (is (thrown? Exception ((init-with-db "test-db" {:connection-migrator-username con-mig-user
```

Comment:
I think there is an extra set of parens wrapping the **init-with-db** call.

Also, might be worth using **(is (thrown-with-msg? Exception #"everything ok exception" ...))** In the assertion here. It probably would never happen, but if **init-with-db** somehow threw an Exception before it got to the assertions in the mocked **make-connection-pool** func I think the test would still pass.

Reply:
* removed the extra parens
* added check for exception message